umap/static/umap/js/umap.features.js

```diff
@@ -15,21 +15,29 @@ U.FeatureMixin = {
   onCommit: function () {
     // When the layer is a remote layer, we don't want to sync the creation of the
     // points via the websocket, as the other peers will get them themselves.
-    if (this.datalayer.isRemoteLayer()) return
+    if (this.datalayer && this.datalayer.isRemoteLayer()) return
+
+    // The "endEdit" event is triggered at the end of an edition,
+    // and will trigger the sync.
+    // In the case of a deletion (or a change of layer), we don't want this
+    // event triggered to cause a sync event, as it would reintroduce
+    // deleted features.
+    // The `._marked_for_deletion` private property is here to track this status.
+    if (this._marked_for_deletion == true) {
+      this._marked_for_deletion == false
+      return
+    }
     this.sync.upsert(this.toGeoJSON())
   },
 
   getGeometry: function () {
     return this.toGeoJSON().geometry
   },
 
-  syncDelete: function () {
-    this.sync.delete()
-  },
-
   initialize: function (map, latlng, options, id) {
     this.map = map
     this.sync = map.sync_engine.proxy(this)
+    this._mark_for_deletion = true
```

Comment:
I don't get why this is initialized to `true`.

Reply:
That's an error. Fixed.